### DISCUSSION TITLE
I need some help getting started

### DISCUSSION THREAD

**Discussion by shan-mcarthur** [moononournation/Arduino_GFX#281] [General]
I have spent 2 days trying to figure this out and could use some help with getting started.

I have an ESP32 board (NodeMCU-32S) and an ILI9488 TFT board.  See http://www.lcdwiki.com/3.5inch_SPI_Module_ILI9488_SKU:MSP3520

I have attached it with the following:

`/******************************
   wiring of IFI9488 to ESP32 dev board:
      
      IFI9488       ESP32

      VCC           3V3
      GND           GND
      CS            D15   (GPIO15)
      RESET         D4    (GPIO4)
      DC/RS         D2    (CS)
      SDI (MOSI)    D23   (MOSI)
      SCK           D18   (SCK)
      LED           3V3
      SDO (MISO)    D19   (MISO)

******************************/`

My code is the hello world example, but I just can't get it to display anything.  

I have even tried simplifying it to the following:

`#include <Arduino_GFX_Library.h>
Arduino_DataBus *bus = new Arduino_HWSPI(2 /* DC */, 15 /* CS */);
Arduino_GFX *gfx = new Arduino_ILI9488(bus, 4, 0 /* rotation */, false /* IPS */);

void setup(void)
{
    gfx->begin();
    gfx->fillScreen(BLACK);
    gfx->setCursor(10, 10);
    gfx->setTextColor(RED);
    gfx->println("Hello World!");
}

void loop()
{
}
`

My protocol analyzer shows data going to MOSI and I see that big block of 0x00 for the black fill.  I just don't see anything on the TFT panel.  It is lit up because the LED pin is on 3.3v.

The Reset pin is always high.  The DC pin goes low occasionally, and the CS pin goes low for every bit but is high otherwise.

I don't have anything else on those pins or using SPI.  One thing I have picked up elsewhere is that the MCU might not have sufficient power and I might need to provide additional 3.3v power on the side.  Would that cause the board to not show anything?  Or am I missing some basic config steps in the super-simple hello world example?

**Comment by moononournation** [comment]
For SPI, you need to use Arduino_ILI9488_18bit

**Reply by shan-mcarthur** [reply]
Bingo!  That worked beautifully.  I can now see my hello world.  Out of curiosity, if the required class is Arduino_ILI9488_18bit, why do we have an Arduino_ILI9488 class that doesn't work?  Is there another reason for this class?

Thanks again for the assist!  I blew an entire weekend on this and didn't realize that I was so close.

**Reply by shan-mcarthur** [reply]
Looking more into the 18-bit.  I was thinking that it was standard RGB, which is 8 bit per color or 24 bit total.  I assume that the 18bit display class used above is specifying the color definition of 18 bit and not a bus width.  ie: colors being defined as 6-bit instead of 8-bit.  So total each color can have 64 values instead of 256 and a total combination of 262K colors instead of the full gammit of 16.7M colors.  Have I understood this correctly?

Is this a restriction of the board, or just a current state of the library?  I am looking at the datasheet for the ILI9488 and it does support full color mode.  I would like to understand what is driving this limitation.  It definately will increase speed but at the cost of converting image files to a different color set which just complicates the dev inner loop and will limit end-user-provided content (assuming I build that in).

**Reply by moononournation** [reply]
16-bit color only supported in parallel interfaces.
